File: apps/ios/ADHDFocus/Models/Project.swift
import Foundation

struct Project: Identifiable, Codable {
    let id: String
    var name: String
    var description: String?
    var emoji: String?
    var color: String?
    var archived: Bool
    var taskCount: Int?
    var completedCount: Int?
    var createdAt: String
    var updatedAt: String
}

struct CreateProjectInput: Codable {
    var name: String
    var description: String?
    var emoji: String?
    var color: String?
}

File: apps/mobile/types/project.ts
export interface Project {
  id: string;
  userId: string;
  name: string;
  description: string | null;
  color: string | null;
  emoji: string | null;
  archived: boolean | null;
  createdAt: string | null;
  updatedAt: string | null;
  // Computed by API (JOIN)
  taskCount?: number;
  completedCount?: number;
}
